services/docker/docker-helpers.js

```diff
@@ -5,12 +5,42 @@ const dockerBlue = '066da5' // see https://github.com/badges/shields/pull/1690
 function buildDockerUrl(badgeName) {
   return {
     base: `docker/${badgeName}`,
-    pattern: ':user/:repo',
+    pattern: ':user/:repo/:tag*',
   }
 }
 
 function getDockerHubUser(user) {
   return user === '_' ? 'library' : user
 }
 
-module.exports = { dockerBlue, buildDockerUrl, getDockerHubUser }
+async function getMultiPageData({ user, repo, fetch }) {
```

Comment:
This appears to be working correctly based on some cursory testing, and it's fairly response (even for `node`) all things considered, so nicely done! 

One thing that we should update here are the comments. What we really want to have within inline comments is specific context, why something's being done, etc.  but we don't need to describe what individual lines of code are doing (the code itself is easy to follow). 

As an example, it'd be good to have a comment that would convey to any future readers/debugers of this code that the API endpoint has a max page size of 100 (explaining where that `100` comes from on line 20 so it doesn't seem like a magic number) but a lot of the popular images will have well over 100 results. 

I think we could also tweak the code to be a bit more succinct (for example leveraging `Promise.all`). I believe something like the below would achieve the same result (though it may need to be tweaked/prettier formatted/etc.)

```js
async function getMultiPageData({ user, repo, fetch }) {
  const data = await fetch({ user, repo })
  const numberOfPages = Math.ceil(data.count / 100)

  if (!numberOfPages) {
    return data.results
  }

  const pageData = await Promise.all([...Array(numberOfPages - 1).keys()].map((_, i) => {
    return fetch({ user, repo, page: ++i + 1 })
  }))
  return [ ...data.results ].concat(...pageData.map(p => p.results))
}
```

Reply:
Understood and will update accordingly.